include/tvm/runtime/container.h

```diff
@@ -274,6 +276,137 @@ class ADT : public ObjectRef {
   TVM_DEFINE_OBJECT_REF_METHODS(ADT, ObjectRef, ADTObj);
 };
 
+/*! \brief An object representing string. It's POD type. */
+class StringObj : public Object {
+ public:
+  /*! \brief The length of the string object. */
+  uint32_t size;
+
+  /*! \brief The pointer to string data. */
+  const char* data;
+
+ private:
+  /*! \brief String object which is moved from std::string container. */
+  class FromStd;
+
+  friend class String;
+};
+
+/*! \brief reference to string objects. */
+class String : public ObjectRef {
+ public:
+  /*!
+   * \brief Construct a new String object
+   *
+   * \param other The moved/copied std::string object
+   *
+   * \note If user passes const reference, it will trigger copy. If it's rvalue,
+   * it will be moved into other.
+   */
+  inline explicit String(std::string other);
+
+  /*!
+   * \brief Compare is equal to other std::string
+   *
+   * \param other The other string
+   */
+  inline bool operator==(std::string other) const;
+
+  /*!
+   * \brief Compare is not equal to other std::string
+   *
+   * \param other The other string
+   */
+  inline bool operator!=(std::string other) const;
+
+  /*!
+   * \brief Compare is equal to other char string
+   *
+   * \param other The other char string
+   */
+  inline bool operator==(const char* other) const;
+
+  /*!
+   * \brief Compare is not equal to other char string
+   *
+   * \param other The other char string
+   */
+  inline bool operator!=(const char* other) const;
+
+  /*!
+   * \brief Returns a pointer to the char array in the string.
+   *
+   * \return const char*
+   */
+  inline const char* c_str() const;
+
+  /*!
+   * \brief Return the length of the string
+   *
+   * \return size_t string length
+   */
+  inline size_t size() const;
+
+  /*!
+   * \brief Return the data pointer
+   *
+   * \return const char* data pointer
+   */
+  inline const char* data() const;
+
+  /*!
+   * \brief Convert String to an std::sting object
+   *
+   * \return std::string
+   */
+  inline operator std::string() const;
+
+  TVM_DEFINE_OBJECT_REF_METHODS(String, ObjectRef, StringObj);
+};
+
+/*! \brief An object representing string moved from std::string. */
+class StringObj::FromStd : public StringObj {
+ public:
+  /*! \brief Container that holds the memory. */
+  std::string data_container;
+};
+
+inline String::String(std::string other) {
```

Comment:
```suggestion
inline String::String(std::string&& other) {
```
I think using rvalue ref here is safer as `other` will be cleared in this constructor.

Reply:
Making it rvalue will require user of the API to always move other, making it value will enable coping `other` if user wants to keep the string.